mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala

```diff
@@ -566,26 +581,41 @@ object LocalLDAModel extends MLReadable[LocalLDAModel] {
     }
   }
 
+  private case class Data(
+                          vocabSize: Int,
+                          topicsMatrix: Matrix,
+                          docConcentration: Vector,
+                          topicConcentration: Double,
+                          gammaShape: Double)
+
   private class LocalLDAModelReader extends MLReader[LocalLDAModel] {
 
     private val className = classOf[LocalLDAModel].getName
 
     override def load(path: String): LocalLDAModel = {
+      // Import implicits for Dataset Encoder
+      val sparkSession = super.sparkSession
+      import sparkSession.implicits._
+      // Pattern to determine sparkversion
+      val versionRegex = """\\d+.\\d+(.\\d+)?(-SNAPSHOT)?""".r
+
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.parquet(dataPath)
-        .select("vocabSize", "topicsMatrix", "docConcentration", "topicConcentration",
-          "gammaShape")
-        .head()
-      val vocabSize = data.getAs[Int](0)
-      val topicsMatrix = data.getAs[Matrix](1)
-      val docConcentration = data.getAs[Vector](2)
-      val topicConcentration = data.getAs[Double](3)
-      val gammaShape = data.getAs[Double](4)
+      val vectorConverted = MLUtils.convertVectorColumnsToML(data, "docConcentration")
+      val Row(vocabSize: Int, topicsMatrix: Matrix, docConcentration: Vector,
+        topicConcentration: Double, gammaShape: Double) = MLUtils.convertMatrixColumnsToML(
+        vectorConverted, "topicsMatrix").select("vocabSize", "topicsMatrix", "docConcentration",
+        "topicConcentration", "gammaShape").head()
       val oldModel = new OldLocalLDAModel(topicsMatrix, docConcentration, topicConcentration,
         gammaShape)
       val model = new LocalLDAModel(metadata.uid, vocabSize, oldModel, sparkSession)
-      DefaultParamsReader.getAndSetParams(model, metadata)
+      metadata.sparkVersion match {
+        case "1.6" =>
+          model.extractTopicDistributionCol(metadata, model)
+        case versionRegex =>
```

Comment:
This is not doing regex pattern matching.  It is creating a temp val "versionRegex" storing the value of metadata.sparkVersion; it matches on everything.

I would recommend using a regex to extract the ```\d+\.\d+``` major and minor version pattern and then doing a match-case on that extracted value.  You could create a helper method for extracting the version so that you do not duplicate code.